etc/flight-cert.yaml

```diff
@@ -44,60 +41,55 @@ config :program_description,  default: 'Manage the HTTPs server and SSL certific
+# cert_type: 'lets_encrypt'
 
 # ==============================================================================
 # Email
 # Stores the email address that all newly generated certificates will be
-# associated with
+# associated with.
 #
 # The email is required for lets-encrypt certificates but optional for
-# self-signed
+# self-signed.
 # ==============================================================================
-config :email, required: false
+# email:
 
 # ==============================================================================
 # Domain
 # Stores the domain all newly generated certificates will be associated with.
 # ==============================================================================
-config :domain, required: false
+# domain:
 
 # ==============================================================================
 # Self Signed Directory
 # Define the directory to where self signed certificates will be stored. This
 # path should not be used by the external web service.
 #
-# It must be an absolute path.
+# Relative paths are expanded from the install directory.
 # ==============================================================================
-#<% if selfsigned_dir -%><%= selfsigned_dir %><% else -%>
-config :selfsigned_dir, default: File.expand_path('self-signed', __dir__)
-#<% end -%>
+# selfsigned_dir: self-signed
 
 # ==============================================================================
 # SSL Fullchain Certificate and Private Key
 # Define the path to the target SSL fullchain  and private key files. The
 # application will symlink the relevant files to these locations. The external
 # web service should use these symlink files.
 #
-# They must be absolute paths.
+# Relative paths are expanded from the install directory.
 # ==============================================================================
-#<% if ssl_paths -%><%= ssl_paths %><% else -%>
-config :ssl_fullchain,  default: File.expand_path('ssl/fullchain.pem', __dir__)
-config :ssl_privkey,    default: File.expand_path('ssl/privkey.pem', __dir__)
-#<% end -%>
+# ssl_fullchain: ssl/fullchain.pem
+# ssl_privkey: ssl/privkey.pem
 
 # ==============================================================================
 # Lets Encrypt Live Directory
 # Define where LetsEncrypt stores its live certificates.
 #
-# It must be an absolute path.
+# Relative paths are expanded from the install directory.
 # ==============================================================================
-#<% if letsencrypt_live_dir -%><%= letsencrypt_live_dir %><% else -%>
-config :letsencrypt_live_dir, default: '/etc/letsencrypt/live'
-#<% end -%>
+# letsencrypt_live_dir: /etc/letsencrypt/live
 
 # ==============================================================================
 # Certbot "Binary" and Flags
+# XXX What are these "internal security reasons?
 # Define the path to the certbot bin file. For internal security reasons,
 # certbot binary file must be an absolute path.
```

Comment:
@WilliamMcCumstie this comment about "internal security reasons" was copied from the `config.reference` file.  Can you expand on what these are please.

Reply:
The comment maybe a bit melodramatic. On further consideration, I'm not sure if the following qualifies as a security issue.

However when we sub-execute `certbot`, we do so without invoking a subshell:
https://github.com/openflighthpc/flight-cert/blob/8026e70ca3299019ab1cc3a3afae1acf9bec0630/lib/flight_cert/commands/cert_gen.rb#L133

This is done to prevent issue with a `bash` shell expanding the `domain` and `email` inputs. However as a consequence, the path to the `certbot` binary is required.

I'm not sure what happens if a relative path is used, it may preform a `PATH` lookup and run anyway. However in general, it should be the absolute path.

Reply:
Thanks for the response.  FWIW a `PATH` lookup will be performed.